src/coord/src/catalog.rs

```diff
@@ -1937,6 +1937,34 @@ impl Catalog {
     pub fn entries(&self) -> impl Iterator<Item = &CatalogEntry> {
         self.by_id.values()
     }
+
+    /// Returns the items in a "time domain" for an expression.
+    ///
+    /// A "time domain" is a set of relations (tables, views, sources) that need to
+    /// share time guarantees. Currently we assume time domains to mean "everything
+    /// in a schema". For example, a read transaction on table A in some schema may
+    /// also need to query table B in the same schema, so A and B (and all other
+    /// tables and views in the schema) are in the same time domain.
+    pub fn timedomain_for(&self, source: &MirRelationExpr, conn_id: u32) -> Vec<GlobalId> {
+        // Find all relations referenced by the expression. Find their parent schemas
+        // and add all tables and views in those schemas to a set.
```

Comment:
super minor nit but this comment doesn't exactly match the code because you add sources too.

Reply:
Implementation changed after it was written. Fixed.